dev/flipbook.tsx

```diff
@@ -53,7 +55,22 @@ export function Flipbook() {
     const numQuestions = selectNumQuestions(state);
     const index = selectCurrentQuestionIndex(state);
 
-    const noTextEntered = state.questions.trim() === "";
+    const questionsState = state.questions.trim();
+    const noTextEntered = questionsState === "";
+
+    useEffect(() => {
+        const localStorageQuestions = localStorage.getItem(LS_QUESTIONS_KEY);
+        if (noTextEntered && localStorageQuestions) {
+            dispatch(setQuestions(localStorageQuestions));
+        }
+
+        localStorage.setItem(LS_QUESTIONS_KEY, questionsState);
+    }, [noTextEntered, questionsState]);
+
+    function handleDiscardQuestion() {
+        dispatch(removeCurrentQuestion);
+        localStorage.removeItem(LS_QUESTIONS_KEY);
```

Comment:
Is this `removeItem` needed? Discarding a question is supposed to remove just that one question from the list, so I don't think we should clear out the storage.

Reply:
Open to alternatives, but it works like this:
- Dispatch to remove the one question (React state updates are async IIRC, so really we're queuing an update here)
- Remove the data from local storage
- State update causes `questionsState` to change
- The change to `questionsState` causes `useEffect` to rerun
- The effect stores the updated `questionsState` in local storage

Reply:
Makes sense, but, since the updated `questionsState` will ultimately be stored (overwriting whatever state is already there), why do we need to clear it first?

Reply:
I have my reasons!

Just kidding, I didn't need to write that code. Fixed.